aten/src/ATen/NestedTensorImpl.h

```diff
@@ -142,6 +162,32 @@ struct TORCH_API NestedTensorImpl : public c10::TensorImpl {
   c10::intrusive_ptr<TensorImpl> shallow_copy_and_detach_core(
       VariableVersion&& version_counter,
       bool allow_tensor_metadata_change) const;
+
+  /**
+   * Generates a non-nested key_set from a nested tensor.
+   *
+   * For many nested tensor kernel implementations a buffer tensor
+   * is generated and redispatched to a non-nested kernel this function
+   * generates the key set used by that buffer tensor
+   *
+   *@return  New key set:
+   *nt.key_set() - NestedTensor(backend) - NestedAutograd + Dense +
+   *Autograd(backend)
+   *
+   * @return A newly constructed view tensor
+   */
+  inline c10::DispatchKeySet generate_buffer_key_set() const {
```

Comment:
I found it cleaner to build back up the new keyset from the backend Key then trying to make a copy of the keyset and subtract and add the new keys- mostly because of aliasing

Reply:
The problem is that you're loosing all the keys that you didn't special case for here: AMP, functionalization, conj bit, neg bit, etc.
That's why I was suggesting to copy the current keyset and work based off that.

Reply:
Yeah totally agree and thought about this. One part of me likes the fact that we are totally explicit though about what keys we support for nested tensors.  But Ill try again to do this with set differences and additions and see if I can get it be cleanish

Reply:
> we are totally explicit 

The main worry I have about this is that every time we add/remove keys, that mean that we will need a NT expert to let us know if this function needs changing or not :/

Reply:
yeah I get that, agree.

Reply:
I was doing this in a grosser manner before. You were completely right this is the way.  ADInplaceOrView is left untouched so that is why it is not explicitly removed and added.